src/components/HistoricalWeatherForm/HistoricalWeatherForm.tsx

```diff
@@ -64,17 +63,18 @@ export default function HistoricalWeatherForm({
           css={classes.form}>
           <Field
             component={SelectField<LanguageOption>}
-            label={LANGUAGE_CHOICE_LABEL}
+            label={t('labels.languageSelect')}
             labelId={LANGUAGE_CHOICE_LABEL_ID}
             name='language'
+            onSelectLanguage={onSelectLanguage}
```

Comment:
Rename it. The `SelectField` is generic, it shouldn't be aware about languages. Also, don't use `Languages` interface in the component